Error when calling Disconnect on already closed connection

Hey,

I'm getting the following error when calling `pingclient.Disconnect()` when the connection was already closed.

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x668b98]

goroutine 80565 [running]:
github.com/xrjr/mcutils/pkg/ping.(*PingClient).Disconnect(0xc000ef5f68)
        /go/src/gocode/vendor/github.com/xrjr/mcutils/pkg/ping/client.go:197 +0x18
main.retrieveMinecraft({0xc000239734, 0x66c38d}, 0x0)
        /go/src/gocode/cmd/main/main.go:114 +0x2f7
created by main.queryServers.func1
        /go/src/gocode/cmd/main/main.go:182 +0x1ef
 ```

--- a/pkg/ping/client.go
+++ b/pkg/ping/client.go
@@ -194,6 +194,10 @@ func (client *PingClient) Ping() (int, error) {
 // Disconnect closes the connection.
 // Connection is made not usable anymore no matter if the it closed properly or not.
 func (client *PingClient) Disconnect() error {
+	if client.conn == nil {
+		return networking.ErrConnectionNotEstablished
+	}
+
 	err := client.conn.Close()
 	client.conn = nil
 	return err

--- a/pkg/ping/legacy.go
+++ b/pkg/ping/legacy.go
@@ -241,6 +241,10 @@ func (client *PingClientLegacy) ping(use1_6_4protocol bool) (LegacyPingInfos, in
 // Disconnect closes the connection.
 // Connection is made not usable anymore no matter if the it closed properly or not.
 func (client *PingClientLegacy) Disconnect() error {
+	if client.conn == nil {
+		return networking.ErrConnectionNotEstablished
+	}
+
 	err := client.conn.Close()
 	client.conn = nil
 	return err

--- a/pkg/query/client.go
+++ b/pkg/query/client.go
@@ -335,6 +335,10 @@ func (client *QueryClient) FullStat(challengeToken uint32) (FullStat, error) {
 // Disconnect closes the connection.
 // Connection is made not usable anymore no matter if the it closed properly or not.
 func (client *QueryClient) Disconnect() error {
+	if client.conn == nil {
+		return networking.ErrConnectionNotEstablished
+	}
+
 	err := client.conn.Close()
 	client.conn = nil
 	return err

--- a/pkg/rcon/client.go
+++ b/pkg/rcon/client.go
@@ -262,6 +262,11 @@ func (client *RCONClient) Command(command string) (string, error) {
 // Connection is made not usable anymore no matter if the it closed properly or not.
 func (client *RCONClient) Disconnect() error {
 	client.authenticated = false
+
+	if client.conn == nil {
+		return networking.ErrConnectionNotEstablished
+	}
+
 	err := client.conn.Close()
 	client.conn = nil
 	return err

